docs/tutorials/optimize/memory/memory_intro.md

```diff
@@ -1,5 +1,56 @@
 ## Memory Optimization
 
-[intro to the pages below]
+Beginning with Mbed OS 5, new features such as RTOS created an increase in flash and RAM usage. This guide explains how to optimize program memory usage for release builds using Mbed OS 5.
 
-[A generic document about writing code and optimizations for mbed OS. We should incorporate this blog: https://developer.mbed.org/blog/entry/Optimizing-memory-usage-in-mbed-OS-52/.]
+**Note:** More information about the memory usage differences between Mbed OS 2 and Mbed OS 5 is available [here](https://developer.mbed.org/blog/entry/Optimizing-memory-usage-in-mbed-OS-52/).
+
+### Removing Unused Modules
+
+For a simple program like [Blinky](https://github.com/ARMmbed/mbed-os-example-blinky), a program that flashes an LED, typical memory usage is split between the following modules: 
+
+```
++---------------------+-------+-------+-------+
+| Module              | .text | .data |  .bss |
++---------------------+-------+-------+-------+
+| Fill                |   132 |     4 |  2377 |
+| Misc                | 28807 |  2216 |    88 |
+| features/frameworks |  4236 |    52 |   744 |
+| hal/common          |  2745 |     4 |   325 |
+| hal/targets         | 12172 |    12 |   200 |
+| rtos/rtos           |   119 |     4 |     0 |
+| rtos/rtx            |  5721 |    20 |  6786 |
+| Subtotals           | 53932 |  2312 | 10520 |
++---------------------+-------+-------+-------+
+```
+
+Even though we are creating a release build, the `features/frameworks` module includes the Mbed OS test tools. Because of this, we are building one of our test harnesses into every binary. Since we do not need a testing framework for a release build, [removing this module](https://github.com/ARMmbed/mbed-os/pull/2559) will save a significant amount of RAM and flash memory.
+
+#### Printf and UART
+
+Other modules that are not used in your program can also be removed by the linker. For example, in [Blinky](https://github.com/ARMmbed/mbed-os-example-blinky) neither `printf` or UART drivers are used within its `main` program. However, every Mbed OS module handles traces and assertions by redirecting their error messages to `printf` on serial output - forcing the `printf` and UART drivers to be compiled in, requiring a large amount of flash memory.
```

Comment:
- Please change first sentence to "The linker can also remove other modules that your program does not use." for active voice.
- Please change second sentence to "For example, [Blinky's main program](https://github.com/ARMmbed/mbed-os-example-blinky) doesn't use `printf` or UART drivers." for active voice.
- Query: In the last sentence, could we change "compiled in" to "compiled", or would that change the meaning?
- In the last sentence, please change "compiled in, requiring a large" to "compiled in and requiring a large" for clarity.